src/Controls/src/Build.Tasks/CompiledConverters/FontSizeTypeConverter.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Maui.Controls.Build.Tasks;
+using Microsoft.Maui.Controls.Xaml;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Microsoft.Maui.Controls.XamlC
+{
+	class FontSizeTypeConverter : ICompiledTypeConverter
+	{
+		public IEnumerable<Instruction> ConvertFromString(string value, ILContext context, BaseNode node)
+		{
+			var module = context.Body.Method.Module;
+
+			if (!string.IsNullOrEmpty(value))
+			{
+				value = value.Trim();
+				if (double.TryParse(value, NumberStyles.Number, CultureInfo.InvariantCulture, out double size))
+				{
+					yield return Instruction.Create(OpCodes.Ldc_R8, size);
+					yield break;
+				}	
+			
+				if (Enum.TryParse<NamedSize>(value, out NamedSize namedSize))
+				{
+					//Device.GetNamedSize(namedSize, targetObject)
+					yield return Instruction.Create(OpCodes.Ldc_I4, (int)namedSize);
+					var parent = node.Parent as IElementNode;
+					if (parent != null && context.Variables.ContainsKey(parent))
+					    yield return Instruction.Create(OpCodes.Ldloc, context.Variables[parent]);
+					else
+					    yield return Instruction.Create(OpCodes.Ldnull);
```

Comment:
It looks like Device.GetNamedSize is going to NRE on this:

[https://github.com/dotnet/maui/blob/e37b7bb2445348622e22c516044ac4c537d18f6e/src/Controls/src/Core/Device.cs#L120-L123](https://github.com/dotnet/maui/blob/e37b7bb2445348622e22c516044ac4c537d18f6e/src/Controls/src/Core/Device.cs#L120-L123)

Reply:
yep, and it would throw a TargetInvocationException if the element isn't a `Element`. I realised that during the night...

Reply:
fixed by using the other overload, so it's a 1:1 match with the non-compiled version